proto/raft_serverpb.proto

```diff
@@ -60,6 +60,53 @@ message SnapshotChunk {
 
 message Done {}
 
+message TabletSnapshotFileMeta {
+    uint64 file_size = 1;
+    string file_name = 2;
+    // Some block data.
+    bytes head_chunk = 3;
+    // trailing data including checksum.
+    bytes trailing_chunk = 4;
+}
+
+message TabletSnapshotPreview {
+    repeated TabletSnapshotFileMeta meta = 1;
+    // There may be too many metas, use a flag to indicate all metas
+    // are sent.
+    bool end = 2;
+}
+
+message TabletSnapshotFileChunk {
+    uint64 file_size = 1;
+    string file_name = 2;
+    // data does not include the first chunk sent in `TabletSnapshotPreview`.
+    bytes data = 3;
```

Comment:
Is the data has self-sufficient data integrity mechanism? If not, I think it's better to include the checksum inside this struct. Relying the caller to store checksum somewhere is not as handy.

Reply:
Now a checksum will be computed for the whole snapshot and included in the `TabletSnapshotEnd` message.